modules/checklist/steps-manager.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Elementor\Modules\Checklist;
+
+use Elementor\Modules\Checklist\Steps\Create_Pages;
+use Elementor\Modules\Checklist\Steps\Step_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Steps_Manager {
+	/** @var Step_Base[] $step_instances */
+	private array $step_instances = [];
+
+	private Module $module;
+
+	public function __construct( Module $module ) {
```

Comment:
step manager can be decoupled it if will get an interface of the module and not the actual module class in the CTOR

Reply:
Done